pkg/devicerepository/store/remote/schema.go

```diff
@@ -0,0 +1,460 @@
+		LoRaWANPHYVersion: phyVersion,
+	}
+
+	if info.CodecID != "" {
+		dev.Formatters = &ttnpb.MessagePayloadFormatters{
+			DownFormatter: ttnpb.PayloadFormatter_FORMATTER_REPOSITORY,
+			UpFormatter:   ttnpb.PayloadFormatter_FORMATTER_REPOSITORY,
+		}
+		paths = append(paths, "formatters")
+	}
+
+	dev.MACSettings = &ttnpb.MACSettings{}
+	if p.ClassBTimeout > 0 {
+		t := time.Duration(p.ClassBTimeout) * time.Second
+		dev.MACSettings.ClassBTimeout = &t
+		paths = append(paths, "mac_settings.class_b_timeout")
+	}
+	if p.ClassCTimeout > 0 {
+		t := time.Duration(p.ClassCTimeout) * time.Second
+		dev.MACSettings.ClassCTimeout = &t
+		paths = append(paths, "mac_settings.class_c_timeout")
+	}
+	if p.PingSlotDataRateIndex > 0 {
+		dev.MACSettings.PingSlotDataRateIndex = &ttnpb.DataRateIndexValue{
+			Value: p.PingSlotDataRateIndex,
+		}
+		paths = append(paths, "mac_settings.ping_slot_data_rate_index")
+	}
+	if p.PingSlotFrequency > 0 {
+		dev.MACSettings.PingSlotFrequency = &pbtypes.UInt64Value{
+			Value: uint64(p.PingSlotFrequency * 100000),
+		}
+		paths = append(paths, "mac_settings.ping_slot_frequency")
+	}
+	if p.PingSlotPeriod > 0 {
+		dev.MACSettings.PingSlotPeriodicity = &ttnpb.PingSlotPeriodValue{
+			Value: pingSlotPeriodToPB[p.PingSlotPeriod],
+		}
+		paths = append(paths, "mac_settings.ping_slot_periodicity")
+	}
+	if p.Rx1Delay > 0 {
+		dev.MACSettings.Rx1Delay = &ttnpb.RxDelayValue{
+			Value: p.Rx1Delay,
+		}
+		paths = append(paths, "mac_settings.rx1_delay")
+	}
+	if p.Rx1DataRateOffset > 0 {
+		dev.MACSettings.Rx1DataRateOffset = &pbtypes.UInt32Value{
+			Value: p.Rx1DataRateOffset,
+		}
+		paths = append(paths, "mac_settings.rx1_data_rate_offset")
+	}
+	if p.Rx2DataRateIndex > 0 {
+		dev.MACSettings.Rx2DataRateIndex = &ttnpb.DataRateIndexValue{
+			Value: p.Rx2DataRateIndex,
+		}
+		paths = append(paths, "mac_settings.rx2_data_rate_index")
+	}
+	if p.Rx2Frequency > 0 {
+		dev.MACSettings.Rx2Frequency = &pbtypes.UInt64Value{
+			Value: uint64(p.Rx2Frequency * 100000),
+		}
+		paths = append(paths, "mac_settings.rx2_frequency")
+	}
+	if p.Supports32BitFCnt {
+		dev.MACSettings.Supports32BitFCnt = &pbtypes.BoolValue{
+			Value: true,
+		}
+		paths = append(paths, "mac_settings.supports_32_bit_f_cnt")
+	}
+	if fs := p.FactoryPresetFrequencies; len(fs) > 0 {
+		dev.MACSettings.FactoryPresetFrequencies = make([]uint64, 0, len(fs))
+		for _, freq := range fs {
+			dev.MACSettings.FactoryPresetFrequencies = append(dev.MACSettings.FactoryPresetFrequencies, uint64(freq*100000))
+		}
+		paths = append(paths, "mac_settings.factory_preset_frequencies")
+	}
+	if dc := p.MaxDutyCycle; dc > 0 {
+		dev.MACSettings.DesiredMaxDutyCycle = &ttnpb.AggregatedDutyCycleValue{
+			Value: dutyCycleFromFloat(dc),
+		}
+		paths = append(paths, "mac_settings.desired_max_duty_cycle")
+	}
+
+	dev.MACState = &ttnpb.MACState{
+		DesiredParameters: ttnpb.MACParameters{},
+	}
+	if p.MaxEIRP > 0 {
+		dev.MACState.DesiredParameters.MaxEIRP = p.MaxEIRP
```

Comment:
This does not persist between sessions - currently NS uses either FP or band defaults, @johanstokking should we add a MAC setting for it?